### PR TITLE
Finalize Job Index Page

### LINKED ISSUE #2340: Task - HR - Finalize Functionality of Job Index page
# Description
Put together the pieces of the Job Index page for HR advisers and make sure they work. This includes communicating with server.

- [ ] This task requires a unit test.

### PR DESCRIPTION
**Pull request by tristan-orourke**
Resolves #2340.

### Notes:
- I tried out Redux's Hooks api! I think it makes a lot of sense for a root component like this, that just has to manage loading/filtering all the data. Apparently it can't cache child components the same way the `connect` HOC can, so it wouldn't be useful in other cases (eg our current Assessment Plan builder, which uses redux extensively to avoid re-rendering child components when parents are re-rendered).
- I made `localizeField` helper function. I've tried doing something like this before, but I finally understand how to use Typescript to make this function with useful typechecking. I know we have the ongoing translation package refactor. Is it too late to be useful?
- I also made a `readableDateTime` helper function. It just leverages dayjs, but it keeps the logic all in one place. We probably want date- or time-exclusive functions later. Localization included!
- There's not much for loading indicators yet. We just show the null states while jobs are loading.

### CODE REVIEW

File: resources/assets/js/helpers/routes.ts [hunk 1]
```diff
@@ -75,3 +75,19 @@ export function managerFaq(locale: string, faqSection?: FaqSection): string {
   }
   return base;
 }
+
+export function hrJobIndex(locale: string): string {
+  return `/${locale}/hr-adviser/jobs`;
```
Comment (Khristophor) [inline resources/assets/js/helpers/routes.ts]: Should do a global find-replace `adviser` > `advisor` just to ensure consistency

File: resources/assets/js/models/jobUtil.ts [hunk 2]
```diff
@@ -99,3 +102,22 @@ export const getSkillLevelName = (
   }
   return skillLevelName(skill_level_id, skill_type_id);
 };
+
+// TODO: allow for Complete status.
+export const jobStatus = (job: Job): JobStatus => {
```
Comment (Khristophor) [inline resources/assets/js/models/jobUtil.ts]: Shouldn't the server be deciding this?

File: resources/assets/js/helpers/dates.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import dayjs from "dayjs";
+import LocalizedFormat from 'dayjs/plugin/localizedFormat';
+import "dayjs/locale/fr";
+import { Locales } from "./localize";
+
+dayjs.extend(LocalizedFormat)
+
+export const readableDateTime = (locale: Locales, date: Date): string => {
```
Comment (Khristophor) [inline resources/assets/js/helpers/dates.ts]: How are we handling timezones with this implementation? Everything from the server would be UTC, correct?

**Comment by Khristophor** [comment]
@tristan-orourke only other thing I noticed is we're missing a `relinquish` button on the claimed job card, not sure if that's for another story or not.

File: resources/assets/js/helpers/routes.ts (hunk 1 above)
Comment (tristan-orourke) [reply resources/assets/js/helpers/routes.ts]: Ah, you're right, or the other way around. I'll ask about a preferred spelling.

File: resources/assets/js/models/jobUtil.ts (hunk 2 above)
Comment (tristan-orourke) [reply resources/assets/js/models/jobUtil.ts]: Yeah, you're right. We'll be changing the status system more in a future sprint, but for now I could just add an accessor to the JobPoster model that returns this status id using the same logic. Do you think that's okay for now?

File: resources/assets/js/helpers/dates.ts (hunk 3 above)
Comment (tristan-orourke) [reply resources/assets/js/helpers/dates.ts]: Good point, timezones are still ignored here.

File: resources/assets/js/models/jobUtil.ts (hunk 2 above)
Comment (Khristophor) [reply resources/assets/js/models/jobUtil.ts]: If we aren't yet keeping the `job_poster_status` table in sync with the `JobPoster` model's `status()` method, then yes, we should add something with this logic on the PHP side to be consumed by Javascript.

File: resources/assets/js/helpers/dates.ts (hunk 3 above)
Comment (Khristophor) [reply resources/assets/js/helpers/dates.ts]: We should be able to [determine the timezone on the fly](https://stackoverflow.com/a/34602679) (if we want to display based on browser/system info), or we can display an explicit timezone like we're doing on the Twig side.

**Comment by tristan-orourke** [comment]
> @tristan-orourke only other thing I noticed is we're missing a `relinquish` button on the claimed job card, not sure if that's for another story or not.

It doesn't show up on @joshdrink 's [mock-up](https://joshbeveridge.ca/tc-manager-dash/hr-adviser/), I think it was decided to only put that button on the summary page.

**Comment by joshdrink** [comment]
> 
> 
> > @tristan-orourke only other thing I noticed is we're missing a `relinquish` button on the claimed job card, not sure if that's for another story or not.
> 
> It doesn't show up on @joshdrink 's [mock-up](https://joshbeveridge.ca/tc-manager-dash/hr-adviser/), I think it was decided to only put that button on the summary page.

I indeed believe this is true :)

File: resources/assets/js/helpers/dates.ts (hunk 3 above)
Comment (tristan-orourke) [reply resources/assets/js/helpers/dates.ts]: Whoah! dayjs seems to be converting it from utc to my browsers local timezone, on its own!

Comment (tristan-orourke) [reply resources/assets/js/helpers/dates.ts]: Could timezones actually be easy, for once??

Comment (Khristophor) [reply resources/assets/js/helpers/dates.ts]: The question is, do we want that? In most other places we're displaying EST/EDT explicitly haha.

Comment (tristan-orourke) [reply resources/assets/js/helpers/dates.ts]: > The question is, do we want that?
😭

Comment (tristan-orourke) [reply resources/assets/js/helpers/dates.ts]: I'll try to get the right EST/etc code to display explicitly.

Comment (tristan-orourke) [reply resources/assets/js/helpers/dates.ts]: Sounds like Gray thinks local time is fine.

**Pull request merged**